# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gcc/hotfix.patch
# Copyright (C) 2015 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

From 114e4e9470260a839d55aad2421fb646af12697b Mon Sep 17 00:00:00 2001
From: Paul Beeler <pbeeler80@gmail.com>
Date: Mon, 11 May 2015 22:27:25 -0600
Subject: [PATCH] Graphite: Add missing isl headers

Signed-off-by: Paul Beeler <pbeeler80@gmail.com>
---
 gcc/graphite-isl-ast-to-gimple.c | 3 +--
 gcc/graphite-optimize-isl.c      | 1 +
 gcc/graphite-poly.h              | 6 ++++++
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/gcc/graphite-isl-ast-to-gimple.c b/gcc/graphite-isl-ast-to-gimple.c
index 90ce0c6..86a7be2 100644
--- a/gcc/graphite-isl-ast-to-gimple.c
+++ b/gcc/graphite-isl-ast-to-gimple.c
@@ -21,9 +21,8 @@ along with GCC; see the file COPYING3.  If not see
 #include "config.h"
 
 #ifdef HAVE_isl
-#include <isl/set.h>
-#include <isl/map.h>
 #include <isl/union_map.h>
+#include <isl/union_set.h>
 #include <isl/ast_build.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
diff --git a/gcc/graphite-optimize-isl.c b/gcc/graphite-optimize-isl.c
index 8bdf744..d602e67 100644
--- a/gcc/graphite-optimize-isl.c
+++ b/gcc/graphite-optimize-isl.c
@@ -28,6 +28,7 @@ along with GCC; see the file COPYING3.  If not see
 #include <isl/band.h>
 #include <isl/aff.h>
 #include <isl/options.h>
+#include <isl/union_set.h>
 #endif
 
 #include "system.h"
diff --git a/gcc/graphite-poly.h b/gcc/graphite-poly.h
index afc0121..06d91fa 100644
--- a/gcc/graphite-poly.h
+++ b/gcc/graphite-poly.h
@@ -22,6 +22,12 @@ along with GCC; see the file COPYING3.  If not see
 #ifndef GCC_GRAPHITE_POLY_H
 #define GCC_GRAPHITE_POLY_H
 
+#ifdef HAVE_isl
+#include <isl/constraint.h>
+#include <isl/set.h>
+#include <isl/map.h>
+#endif
+
 typedef struct poly_dr *poly_dr_p;
 
 typedef struct poly_bb *poly_bb_p;
